### PR TITLE
Databases Week 2

### PR DESCRIPTION
**Pull request by rebeccahockley**

### CODE REVIEW

File: databases/week2/WorkingWithTasks.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+USE hyfData;
```
Comment (mateusms01) [inline databases/week2/WorkingWithTasks.sql:1]: very good query formatting, clear and readable.

File: databases/week2/MoreQueries.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
+-- Get all the tasks assigned to the user who's email ends in @spotify.com
+SELECT 
+    task.title, user.email
+FROM 
+    task
+JOIN 
+    user
```
Comment (mateusms01) [inline databases/week2/MoreQueries.sql]: Avoid line break on FROM + Table name and on Join + Condition. 
Also, even if obvious the join condition should always be explicitly written.

Comment (mateusms01) [reply databases/week2/MoreQueries.sql]: this goes for all the queries in this file, please revise it,

File: databases/week2/SchoolDatabase.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
+CREATE DATABASE SchoolDatabase;
+
+USE SchoolDatabase;
+
+CREATE TABLE `class` (
+  `Class_id` int unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
```
Comment (mateusms01) [inline databases/week2/SchoolDatabase.sql]: nice use of primary key

File: databases/week2/SchoolDatabase.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  `Name` varchar(100) NOT NULL,
+  `BeginDate` DATE,
+  `EndDate` DATE
+);
+
+CREATE TABLE `student` (
+  `Student_id` int unsigned NOT NULL AUTO_INCREMENT,
```
Comment (mateusms01) [inline databases/week2/SchoolDatabase.sql]: could also have indicated primary key here.

File: databases/week2/LibraryDatabase.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
+CREATE DATABASE LibraryDatabase;
+
+USE LibraryDatabase;
+
+CREATE TABLE `Member` (
+  `Member_ID` int unsigned NOT NULL AUTO_INCREMENT,
```
Comment (mateusms01) [inline databases/week2/LibraryDatabase.sql]: missing primary key indication.
also missing on the other tables.

File: databases/week2/LibraryDatabase.sql [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  `First_Name` varchar(255) NOT NULL,
+  `Last_Name` varchar(255) NOT NULL,
+  `Address` varchar(255) NOT NULL,
+  `Email` varchar(255) NULL,
+  `Phone` varchar(20) NULL
```
Comment (mateusms01) [inline databases/week2/LibraryDatabase.sql:11]: missing constraints statements that indicate the tables relationships through foreign keys.

File: databases/week2/LibraryDatabase.sql [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  `Genre` varchar(255) NOT NULL,
+  `Shelf_ID` int NOT NULL,
+  `Status_ID` int NOT NULL,
+  `Reserved` varchar(20) NOT NULL
+); 
+
+CREATE TABLE `Loaned_Books` (
+  `Index_ID` int unsigned NOT NULL AUTO_INCREMENT,
+  `Member_ID` int unsigned NOT NULL,
+  `Book_ID` int unsigned NOT NULL,
```
Comment (mateusms01) [inline databases/week2/LibraryDatabase.sql:31]: member id and book id should be also indicated as constrains.

File: databases/week2/LibraryDatabase.sql [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  `Borrow_Date` DATETIME NOT NULL,
+  `Due_Date` DATETIME NOT NULL,
```
Comment (mateusms01) [inline databases/week2/LibraryDatabase.sql]:
```suggestion
  `Due_Date` DATETIME NOT NULL,
  CONSTRAINT 'fk_member' FOREIGN KEY ('member_id') REFERENCES 
  'Member'('member_id') ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT 'fk_book' FOREIGN KEY ('Book_ID') REFERENCES 'Book Details'('BookID') ON DELETE CASCADE ON UPDATE CASCADE 
```

**Review by mateusms01 — CHANGES_REQUESTED** [review]
Hey Rebecca, nice work this week, but I would recommend you to re-do your Database task 4 (only this one) and also revise the class content about it. I know you handed in late, and that is completely fine, but this is a quite important topic and would really benefit you if you re-work it on another day when you have more time. 

I would also suggest to give a special attention to relationships definition and how to handle many to many relations, this is something you didn't add in your task 4. 

I would happily review again once you manage to give it a new look.

File: databases/week2/LibraryDatabase.sql (hunk 8 above)
Comment (rebeccahockley) [reply databases/week2/LibraryDatabase.sql]: I get a lot of errors when I run foreign keys, but yes, clearly I have to spend more time on this subject so I will redo the homework.

Comment (mateusms01) [reply databases/week2/LibraryDatabase.sql]: If possible I would recommend bringing this issue to class, it is easier to help you with the error messages in person.

File: databases/week2/MoreQueries.sql [hunk 9]
```diff
@@ -1,40 +1,25 @@
...
-    task.title, user.email
-FROM 
-    task
-JOIN 
-    user
-WHERE
-    user.email LIKE "%@spotify.com";
+task.title, user.email
+FROM task
+INNER JOIN user WHERE user.email LIKE "%@spotify.com";
```
Comment (mateusms01) [inline databases/week2/MoreQueries.sql:6]: Good work re-structuring the code. However my main concern here is that you are not explicitly saying where the inner join should happen. This is something that should always be stated. for example: 

SELECT
task.title,
user.email,
FROM task
INNER JOIN user **ON task.user_id = user.id**
WHERE user.email LIKE "%@spotify.com";

When the join condition is not defined MySQL will choose the join key, which in this case works but its unlikely to work every time.

**Review by mateusms01 — CHANGES_REQUESTED** [review]

**Pull request merged**